src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -753,6 +751,50 @@ private string ConfigureDatabase(InitializeViewRequestParams viewParams, Databas
                             }
                         }
 
+                        if (!viewParams.IsNewObject && database.Filegroups != null)
+                        {
+                            HashSet<string> fileGroupToRemove = new HashSet<string>(prototype.Filegroups.Select(file => file.Name));
+                            // process row data filegroups
+                            foreach (FileGroups fg in database.Filegroups)
+                            {
+                                if (fg.Id < 0)
+                                {
+                                    FilegroupPrototype newfileGroup = new FilegroupPrototype(prototype);
+                                    newfileGroup.FileGroupType = fg.Type;
+                                    newfileGroup.Name = fg.Name;
+                                    newfileGroup.IsReadOnly = fg.IsReadOnly;
+                                    newfileGroup.IsDefault = fg.IsDefault;
+                                    newfileGroup.IsAutogrowAllFiles = fg.IsDefault;
+                                    prototype.Filegroups.Add(newfileGroup);
+                                }
+                                else
+                                {
+                                    var existedFilegroup = prototype.Filegroups.FirstOrDefault(x => x.Name == fg.Name);
```

Comment:
~To avoid n^2 loop iterations here, you could have a separate HashSet that contains all the existing names in the prototype.~

Reply:
Disregard that, we need to get the actual FileGroup object from the prototype. Doing a Dictionary of filegroup Name to FileGroup object would work better there.

Reply:
For example:
```
var groupNameMap = new Dictionary<string, Smo.FileGroup>();
foreach (FileGroup group in prototype.FileGroups)
{
    groupNameMap.Add(group.Name, group)
}

foreach (FileGroups fg in database.Filegroups)
{
    if (groupNameMap.TryGetValue(fg.Name, out existingFileGroup))
    {
        ...
    }
}
```